controllers/k8ssandracluster_controller.go

```diff
@@ -133,6 +140,13 @@ func (r *K8ssandraClusterReconciler) reconcile(ctx context.Context, kc *api.K8ss
 
 			if actualHash, found := actualDc.Annotations[api.ResourceHashAnnotation]; !(found && actualHash == desiredDcHash) {
 				logger.Info("Updating datacenter")
+
+				if actualDc.Spec.SuperuserSecretName != desiredDc.Spec.SuperuserSecretName {
```

Comment:
It's trickier than that. cass-operator's validation will reject any changes to the superUserSecretName. Thus, if we've diverged - the actualDc's value is the one that should be used instead of desired. But if that's different per cluster, then essentially we're pretty screwed.

Thus, whatever you set to K8ssandraCluster object could be something we will never use - and the reconcile would have to use the actualDc value in this case. That's not a pretty sight (we probably need a validation to prevent this in K8ssandraCluster webhook).

Reply:
Thanks for pointing out the cass-operator behavior. It looks like cass-operator prevents you from pointing to a different secret but you could still potentially modify the existing secret.

Two questions,

* Can we wind up in state from which we cannot recover?
* Will existing behavior cause any problems for k8ssandra 1.x migration scenarios?

Reply:
We could, if somehow different CassandraDatacenters use different superuserSecretName. That could happen if user modifies the K8ssandraCluster's SuperuserSecretName and then adds a new Datacenter.

It shouldn't hurt migrations, we already have a superuser there. We need to just copy that setting to our K8ssandraCluster.